classifier.py

```diff
@@ -0,0 +1,129 @@
+from sklearn.metrics import accuracy_score
+import numpy as np
+import random, pickle, argparse, json, os
+
+
+def sampling(lst, class_type, sample_size=500):
+	# given a list, return a sample list of a given size with a class label for
+	# each sample
+	lst_size = len(lst)
+	if sample_size > lst_size:
+		indecies = np.random.choice(range(lst_size),size=sample_size - lst_size)
+		sample_lst = np.vstack((lst,lst[indecies]))
+	else:
+		indecies = random.sample(range(lst_size), sample_size)
+		sample_lst = lst[indecies]
+
+	if class_type == 'positive':
+		labels = np.ones(sample_size, dtype=np.int)
+	elif class_type == 'negative':
+		labels = np.zeros(sample_size, dtype=np.int)
+	else:
+		raise NameError('Unrecognized class type. \
+		       Must be either "positive" or "negative"')
+	labels = labels.reshape(-1, 1)
+	sample = np.hstack((sample_lst, labels))
+	return sample
+
+def balanced_dataset(pos_lst, neg_lst, size):
+	pos_size = size/2
+	neg_size = size - pos_size # in case of odd int value of size
+	pisitives = sampling(pos_lst, 'positive', pos_size)
+	negatives = sampling(neg_lst, 'negative', neg_size)
+	return np.vstack((pisitives,negatives))
+
+
+def search_top_relevant(sim_results, inverse_ndx, sim_th):
+	related = sim_results[:,1] > sim_th
+	related_ndx = []
+	for i, r in enumerate(related):
+		if r:
+			related_ndx.append(i)
+	results = []
+	for i in related_ndx:
+	    results.append(inverse_ndx[str(count_pos + i)])
+	return np.unique(results)
+
+if __name__ == "__main__" :
+	parser = argparse.ArgumentParser()
+	# the file contains the documents decomposed vectors (model) to load
+	parser.add_argument("--vectors", default="results/output_svd.pickle", type=str)
+	parser.add_argument("--inv_ndx", default="results/output_paragraph_inverse_index.json", type=str)
+	parser.add_argument("--count_pos", default=183, type=int)
+	parser.add_argument("--min_relevant", default=5, type=int)
+	parser.add_argument("--training_size", default=1500, type=int)
+	parser.add_argument("--testing_size", default=500, type=int)
+	parser.add_argument("--n_iterations", default=10, type=int)
+	parser.add_argument("--verbose", default=1, type=int)
+	parser.add_argument("--out_dir", default="results", type=str) # name of the output directory
+
+	args = parser.parse_args()
+	vectors = args.vectors
+	inv_ndx = args.inv_ndx
+	count_pos = args.count_pos
+	min_relevant = args.min_relevant
+	training_size = args.training_size
+	testing_size = args.testing_size
+	n_iterations = args.n_iterations
+	verbose = args.verbose
+	out_dir = args.out_dir
+
+	svd_vectors = pickle.load(open(vectors,'rb'))
+	print 'The model (document decomposed vectors) was loaded successfully'
+	vec_size = svd_vectors.shape[1]
+	pos_lst = svd_vectors[:count_pos]
+	neg_lst = svd_vectors[count_pos:]
+
+	scores = np.zeros(n_iterations)
+	for i in range(n_iterations):
+		if verbose > 0:
+			print 'iteration', i + 1 , 'out of ', n_iterations
+		train, test = train_test_split(pos_lst)
```

Comment:
Add the percentage of test set, out of the annotated data. Could be passed as a parameter.